modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rel/agg/IgniteSortAggregateBase.java

```diff
@@ -48,7 +47,22 @@
     @Override default Pair<RelTraitSet, List<RelTraitSet>> passThroughCollation(
         RelTraitSet nodeTraits, List<RelTraitSet> inputTraits
     ) {
-        RelCollation collation = RelCollations.of(ImmutableIntList.copyOf(getGroupSet().asList()));
+        RelCollation required = TraitUtils.collation(nodeTraits);
+        ImmutableBitSet requiredKeys = ImmutableBitSet.of(required.getKeys());
+        RelCollation collation;
+
+        if (getGroupSet().contains(requiredKeys)) {
```

Comment:
```suggestion
        if (getGroupSet().contains(requiredKeys)) {
            ImmutableBitSet keysLeft = getGroupSet().except(requiredKeys);
            
            List<RelFieldCollation> newCollationFields = 
                new ArrayList<>(required.getFieldCollations().size() + keysLeft.size());
            
            newCollationFields.addAll(required.getFieldCollations());
```

Reply:
What is the difference? `requiredKeys` is a subset of `groupSet`, so size of `required + keysLeft = required + (groupSet - required) = groupSet`